File: selection_sort.rb
#Selection sort
require "pry"

def find_smallest(array)
  smallest = array[0]
  smallest_index = 0
  array.each_with_index do |element, index|
    if element < smallest
      smallest = element
      smallest_index = index
    end
  end
  smallest_index
end

def selection_sort(array)
  new_array = []
  while array.length >= 1
    smallest = find_smallest(array)
    new_array.push(array.delete_at(smallest))
  end
  new_array
end

print selection_sort([5, 3, 6, 2, 10])

File: selection_sort.php
<?php

function find_smallest($array) {
  $smallest = $array[0];
  $smallest_index = 0;
  for ($i = 0; $i < count($array); $i++) {
    if ($array[$i] < $smallest) {
      $smallest = $array[$i];
      $smallest_index = $i;
    }
  }
  return $smallest_index;
}

function selection_sort($array) {
  $new_array = [];

  while (count($array) > 0) {
    $smallest = find_smallest($array);
    array_push($new_array, $array[$smallest]);
    array_splice($array, $smallest, 1);
  }

  return $new_array;
}

$array = [5, 3, 6, 2, 10, 1231, 23, 32, 667, 4, 55];
print_r(selection_sort($array));
?>
